tests/common/factories/group.py

```diff
@@ -42,3 +42,18 @@ def scopes(self, create, scopes=0, **kwargs):
             scopes = [GroupScope(group=self) for _ in range(0, scopes)]
 
         self.scopes = scopes or []
+
+
+class RestrictedGroup(Group):
+    name = factory.Sequence(lambda n: 'Test Restricted Group {n}'.format(n=str(n)))
+
+    joinable_by = None
+    readable_by = ReadableBy.world
+    members = []
```

Comment:
Is `joinable_by=None` purposeful here? I had thought restricted groups were joinable by `authority` but only `writeable_by` members?

Reply:
We'll leave it as `None` for now but we may need to change it. The problem is what does "joinable by" mean? Or how exactly does it get used, by what parts of the code?

Our spec for minimal restricted groups right now says that users can't join restricted groups (since we don't even have a design for any type of join or invite process that works for restricted groups), they must be added by admins.

If some code that's involved when an admin adds a member is gonna fail because `joinable_by` is `None` or is gonna require `joinable_by` to be `"authority"` then we'll have to change `joinable_by`.

On the other hand, what about the code for a group's home page (the `/groups/<PUBID>/<SLUG>` pages)? Right now if you visit the home page of a private group that you aren't a member of it will show you a button for joining the group. But if you visit the page of an open or restricted group it'll just show you the group's contents. Is that (whether or not to show a join button) based on `joinable_by`? If so, `joinable_by` will have to be `None` for restricted groups.

We'll have to see how it shakes out I think once we have the admin form and the activity page

Reply:
Aha! We have a different mental model of the meaning of `joinableBy`...but we'll sort it out as we go.